### ISSUE TITLE
Model defaults ignored on empty text field or empty boolean field.

### ISSUE THREAD

**Issue by a1Gupta** [tomchristie/django-rest-framework#3788]
I am using Django 1.8 and DRF 3.3.1. I am still facing similar issue referenced in #1101.

```
class UserSetting(models.Model):
    allow_search_by_email = models.BooleanField(default=True)
    allow_profile_indexing = models.BooleanField(('Allow indexing?'), default=True)

class UserSettingSerializer(serializers.ModelSerializer):
    
    class Meta:
        model = UserPrivacySetting
        fields = [
                    'allow_profile_indexing','allow_search_by_email',
                    ]
```

But whenever I am posting a blank form, and validating the serializer, it sets false to both the fields.

The issue #1101 was closed in #2311 but I am still facing the similar issue.

Is it fixed ? or
Should i use the workaround using extra_kwargs ?

**Comment by xordoquy** [comment]
This should be fixed as we have a passing test case.

**Comment by a1Gupta** [comment]
That means it should take model defaults if field is blank ?

**Comment by xordoquy** [comment]
Somehow there seems to be differences between your use case and the DRF test cases. This difference needs to be highlighted.

**Comment by i-s-o-g-r-a-m** [comment]
@xordoquy When you say this should be fixed, do you mean in an upcoming release? I'm on the latest DRF (3.3.2) and latest Django (1.9) and am still seeing this problem: default values for `BooleanField`s are not working as expected -- but only in the context of unit tests (things work fine via normal access to the API). I was able to use the `extra_kwargs` work-around to set the `default` there, and that fixed the problem I was running into with my tests, but am hoping this might be fixed in the framework. 

Thanks for all the hard work on DRF. It's a lot of fun to use and the docs are great!

**Comment by xordoquy** [comment]
@i-s-o-g-r-a-m #2311 was part of 3.0.3 so it should work with 3.3.2
I'd be really interested in a failing test case or code sample we could look at to understand what's going on.

**Comment by i-s-o-g-r-a-m** [comment]
@xordoquy I spent some time with a fresh Django/DRF install (latest stuff from pip as of 7 Feb 2016) and wasn't able to repro the problem. It's weird. I'm gonna go back to the codebase where I ran into this problem initially to see if I can extract something from there. I'm starting to wonder if the discrepancy might be more occult -- maybe even the backend (I noticed the problem with Postgres, whereas my attempt today was using SQLite).

**Comment by tomchristie** [comment]
> and wasn't able to repro the problem.

Okay, thanks! Closing this until/unless we have a reproducible case.

**Comment by kgodey** [comment]
I ran into this today using DRF 3.3.3 / Django 1.9.6  and Postgres. My Location model has a boolean field called `is_active` with default `True`. The frontend does not send this field to the API.

 When I include `is_active` in the `fields` option of the serializer, it's ignoring the model default and getting saved as `False`. When I remove it from the `fields` option, it's saving as `True`.

**Comment by tomchristie** [comment]
Any chance we could get this reduced to a simple example case?
A description of the serializer, and the request as sent by a tool such as `curl` would be a good point for use to look into reproducing the issue.

**Comment by flaiming** [comment]
I added a simple failing test example for BooleanField. I think the root cause is that serializer is behaving differently when you initialize it with ``request.data`` (which is ``QueryDict``) or just simple dict. Turning point is here: https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/utils/html.py#L12

**Comment by flaiming** [comment]
Could this ticket be reopened please, since we now have failing test code sample?

**Comment by xordoquy** [comment]
Did you made a PR for that failing test case ? We need to investigate it before taking further action

**Comment by tomchristie** [comment]
Keeping this closed, as per note in #4284.

**Comment by resurrexi** [comment]
I was encountering this issue when running my test cases using the `APIClient`.

It turns out that if I send my data as a multipart form request, the test case will fail, but if I specify `format='json'` or configure my settings `'TEST_REQUEST_DEFAULT_FORMAT': 'json'`, my test case will pass.

It might be worth mentioning it in the testing docs.
